librz/core/cfile.c

```diff
@@ -298,8 +312,8 @@ RZ_API void rz_core_file_reopen_debug(RzCore *core, const char *args) {
 		return;
 	}
 	core->dbg->main_arena_resolved = false;
-	RzList *old_sections = __save_old_sections(core);
-	ut64 old_base = core->bin->cur->o->baddr_shift;
+	RzList *sections_backup = rz_core_create_sections_backup(core);
```

Comment:
If you will add a return of NULL in case of error, then you would need to check `sections_backup` for NULL as well.